tests/test_memoryfile.py

```diff
@@ -1,22 +1,22 @@
-"""MemoryFile tests"""
+"""MemoryFile tests.  MemoryFile requires GDAL 2.0+.
+Tests in this file will ONLY run for GDAL >= 2.x"""
 
 from io import BytesIO
 import logging
-
-from packaging.version import parse
 import pytest
 
 import rasterio
 from rasterio.io import MemoryFile, ZipMemoryFile
-
+from rasterio.env import GDALVersion
 
 logging.basicConfig(level=logging.DEBUG)
 
 
-# Custom markers.
-mingdalversion = pytest.mark.skipif(
-    parse(rasterio.__gdal_version__) < parse('2.0dev'),
-    reason="MemoryFile requires GDAL 2.0")
+# Skip ENTIRE module if not GDAL >= 2.x.
+# pytestmark is a keyword that instructs pytest to skip this module.
+pytestmark = pytest.mark.skipif(
```

Comment:
This change allows us to skip the whole module, rather than annotating each test individually